pennylane/templates/embeddings.py

```diff
@@ -123,9 +123,13 @@ def AngleEmbedding(features, wires, rotation='X'):
         raise ValueError("Rotation has to be `X`, `Y` or `Z`; got {}.".format(rotation))
 
 
-def AmplitudeEmbedding(features, wires):
+def AmplitudeEmbedding(features, wires, pad=False):
     r"""Encodes :math:`2^n` features into the amplitude vector of :math:`n` qubits.
 
+    If the total number of features to embed are less than the :math:`2^n` available amplitudes,
+    non-informative constants (zeros) can be padded to ``features``. To enable this, the argument
+    ``pad`` should be set to ``True``. It is set to ``False`` by default. 
+
     The absolute square of all elements in ``features`` has to add up to one.
```

Comment:
Maybe we can also add an option `normalize` that normalizes the features automatically.

Reply:
hmm, yeah. I suppose that is more user-friendly.  

Should I add another argument to the function or just do it internally and print something about it?

Reply:
I'd add it as another argument. And I'd prefer it to be False by default. In this case, users will read the error message and understand that a normalization is necessary. If it would happen by default this could cause some weird behaviour if people don't know that it is happening.

Reply:
I agree but I will wait for @josh146 to comment on this. I am not sure if having too many arguments is generally a good strategy.

Reply:
In this case I think it's fine - I'm starting to think that the benefit of `AmplitudeEncoding`, as opposed to just directly using `QubitStateVector`, is that it provides these useful validations and processing for the user.

Reply:
ok :)